p2p/client_test.go

```diff
@@ -20,16 +21,41 @@ import (
 
 func TestClientStartup(t *testing.T) {
 	privKey, _, _ := crypto.GenerateEd25519Key(rand.Reader)
-	client, err := NewClient(config.P2PConfig{}, privKey, "TestChain", test.NewLogger(t))
 	assert := assert.New(t)
-	assert.NoError(err)
-	assert.NotNil(client)
+	testCases := []struct {
+		desc    string
+		p2pconf config.P2PConfig
+	}{
+		{"blank config", config.P2PConfig{}},
+		{"peer whitelisting", config.P2PConfig{
+			ListenAddress: "",
+			Seeds:         "",
+			BlockedPeers:  "",
+			AllowedPeers:  "/ip4/127.0.0.1/tcp/7676/p2p/12D3KooWM1NFkZozoatQi3JvFE57eBaX56mNgBA68Lk5MTPxBE4U",
+		}},
+		{"peer blacklisting", config.P2PConfig{
+			ListenAddress: "",
+			Seeds:         "",
+			BlockedPeers:  "/ip4/127.0.0.1/tcp/7676/p2p/12D3KooWM1NFkZozoatQi3JvFE57eBaX56mNgBA68Lk5MTPxBE4U",
+			AllowedPeers:  "",
+		}},
+	}
 
-	err = client.Start(context.Background())
-	defer func() {
-		_ = client.Close()
-	}()
-	assert.NoError(err)
+	for _, testCase := range testCases {
+		t.Run(testCase.desc, func(t *testing.T) {
+			gater, err := conngater.NewBasicConnectionGater(nil)
```

Comment:
can you use the in-memory db instead of `nil`? e.g., `conngater.NewBasicConnectionGater(dssync.MutexWrap(datastore.NewMapDatastore()))`
when `nil` is used, it basically skips over all the tests of blocking/unblocking peers.
see https://github.com/libp2p/go-libp2p/blob/master/p2p/net/conngater/conngater.go#L127